suro-kafka-producer/src/main/java/com/netflix/suro/sink/kafka/KafkaSink.java

```diff
@@ -340,12 +330,16 @@ public void call(Throwable throwable) {
     public void close() {
         isOpened = false;
         try {
+            executor.shutdown();
```

Comment:
metadataWaitingQueue.take() will be blocked forever and executor.awaitTermination(10, TimeUnit.SECONDS) will always take 10 seconds. So, we need to set up a timeout with poll(). The example can QueuedSink.java in suro-core. If you want to make things simpler, before calling shutdown(), you can push the 'command' message to metadataWaitingQueue. In the thread loop, if the message is 'command' message, it should skip the message and break the loop.